clients/pkg/promtail/targets/objectstore/client.go

```diff
@@ -0,0 +1,17 @@
+package objectstore
+
+import (
+	"github.com/grafana/loki/pkg/storage/chunk"
+)
+
+type Client interface {
+	ReceiveMessage(timeout int64) ([]messageObject, error)
+	acknowledgeMessage(message interface{}) ackMessage
```

Comment:
That doesn't seems used we should remove it.

Reply:
s3Client implements these two methods. From target we call these methods through the client 
https://github.com/adityacs/loki/blob/object_storage_scrape/clients/pkg/promtail/targets/objectstore/target.go#L91

Reply:
I got your point. `acknowledgeMessage` is not required to be part of `Client`